components/ILIAS/LegalDocuments/classes/ForcedLogoutTarget.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ ********************************************************************
+ */
+
+declare(strict_types=1);
+
+namespace ILIAS\LegalDocuments;
+
+use ilCtrlInterface;
+use ILIAS\components\Authentication\Logout\LogoutTarget;
+use ILIAS\Data\URI;
+use ilStartUpGUI;
+use ILIAS\Authentication\Logout\LogoutDestinations;
+
+class ForcedLogoutTarget implements LogoutTarget
+{
+    public function __construct(
+        private readonly ilCtrlInterface $ctrl,
+        private readonly string $http_path = ILIAS_HTTP_PATH
+    ) {
+    }
+
+    public function asURI(): URI
+    {
+        return LogoutDestinations::LOGIN_SCREEN->asURI($this->ctrl, $this->http_path);
```

Comment:
Actually, this is a little different than my suggested approach.

In my approach, the original "Logout Target" should be passed to the `__construct(...)` of **this** class.

The `asURI` method of **this** class should than decide (with the help of "indicators", also passed to `__construct(...)`), if the URI of the original "Logout Target" is returned, or `LogoutDestinations::LOGIN_SCREEN->asURI($this->ctrl, $this->http_path);`.

Reply:
I disagree with the notion that the `ForcedLogoutTarget` should decide whether the original `$target` should be used or not, the class name does not reflect this. If I see a `ForcedLogoutTarget` I would assume that this **is** the decision that we want to force a specific logout target and not that it would decide if a page should be forced or not.

Regardless, as this is just a minor thing, the current approach is ok for me.